### PR TITLE
[PDI-17853] - Need support to connect with MongoDB Atlas using latest…

### PR DESCRIPTION
**Pull request by GopiKrishnaAngadi**
… SRV connection string

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 3m 7s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl pentaho-mongodb-plugin
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by peterrinehart — CHANGES_REQUESTED** [review]
Looks good overall, but the following changes are necessary:
1) Ensure any existing code files get the copyright dates updated in the header so that the final year listed is 2021.
2) Some of those Sonar code smells should be corrected, such as unused fields, importing using the * operator, etc.  I understand that many of them were already present, but if fixing them doesn't take too much time or add substantial technical risk, let's get them cleaned up too.

**Comment by buildguy** [comment]
### :x: Build failed in 6m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl pentaho-mongodb-plugin
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl pentaho-mongodb-plugin
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java [hunk 1]
```diff
@@ -610,14 +699,14 @@ public void widgetSelected( SelectionEvent arg0 ) {
...
-     * 
+     *
      * @Override public void widgetSelected(SelectionEvent e) {
      * addTagsToTable(); input.setChanged(); } });
-     * 
-     * 
-     * 
+     *
+     *
+     *
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java:709]: If this block of commented out code isn't needed, we should delete it.

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMeta.java [hunk 2]
```diff
@@ -122,7 +133,7 @@ private void readData( Node stepnode ) throws KettleXMLException {
       setAuthenticationDatabaseName( XMLHandler.getTagValue( stepnode, "auth_database" ) ); //$NON-NLS-1$
       setAuthenticationUser( XMLHandler.getTagValue( stepnode, "auth_user" ) ); //$NON-NLS-1$
       setAuthenticationPassword( Encr.decryptPasswordOptionallyEncrypted( XMLHandler.getTagValue( stepnode,
-          "auth_password" ) ) ); //$NON-NLS-1$
+              "auth_password" ) ) ); //$NON-NLS-1$
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMeta.java:136]: Is this spacing consistent with the checkstyle standards?

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDiscoverFields.java [hunk 3]
```diff
@@ -10,13 +11,13 @@
  * Created by brendan on 11/4/14.
  */
 public interface MongoDbInputDiscoverFields {
-  public List<MongoField> discoverFields( MongoProperties.Builder properties, String db, String collection,
-                                          String query, String fields,
-                                          boolean isPipeline, int docsToSample, MongoDbInputMeta step )
+  List<MongoField> discoverFields( MongoProperties.Builder properties, String db, String collection,
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDiscoverFields.java:14]: This file does not appear to have a valid license header.

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*!
- * Copyright 2010 - 2020 Hitachi Vantara.  All rights reserved.
+ * Copyright 2010 - 2021 Hitachi Vantara.  All rights reserved.
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java:2]: If we didn't change anything else in this file, we don't need to update the license header.

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDialog.java [hunk 5]
```diff
@@ -201,7 +216,7 @@ public MongoDbOutputDialog( Shell parent, Object in, TransMeta tr, String name )
     m_stepnameText.setLayoutData( fd );
 
     m_wTabFolder = new CTabFolder( shell, SWT.BORDER );
-    props.setLook( m_wTabFolder, Props.WIDGET_STYLE_TAB );
+    //props.setLook( m_wTabFolder, Props.WIDGET_STYLE_TAB );
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDialog.java:219]: This commented line can be deleted.

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java [hunk 6]
```diff
@@ -640,85 +729,87 @@ public void widgetSelected( SelectionEvent arg0 ) {
...
      * deleteSelectedFromView(); input.setChanged(); } });
      */
     // test tag set but
-    Button testUserTagsBut = new Button( wInputOptionsComp, SWT.PUSH | SWT.CENTER );
+    testUserTagsBut = new Button( wInputOptionsComp, SWT.PUSH | SWT.CENTER );
     props.setLook( testUserTagsBut );
     testUserTagsBut.setText( BaseMessages.getString( PKG, "MongoDbInputDialog.TestUserTags.Button" ) ); //$NON-NLS-1$
     testUserTagsBut.setToolTipText(
-        BaseMessages.getString( PKG, "MongoDbInputDialog.TestUserTags.Button.TipText" ) ); //$NON-NLS-1$
+            BaseMessages.getString( PKG, "MongoDbInputDialog.TestUserTags.Button.TipText" ) ); //$NON-NLS-1$
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java:798]: Is this spacing consistent with style guidelines?

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/mongo/wrapper/MongoWrapperClientFactory.java [hunk 7]
```diff
@@ -8,5 +8,7 @@
  * Created by bryan on 8/22/14.
  */
 public interface MongoWrapperClientFactory {
-  public MongoClientWrapper createMongoClientWrapper( MongoProperties props, MongoUtilLogger log ) throws MongoDbException;
+  MongoClientWrapper createMongoClientWrapper( MongoProperties props, MongoUtilLogger log ) throws MongoDbException;
+
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/mongo/wrapper/MongoWrapperClientFactory.java:12]: This file appears to be missing its license header.

File: pentaho-mongodb-plugin/src/main/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImpl.java [hunk 8]
```diff
@@ -37,6 +37,7 @@
 import org.pentaho.di.core.exception.KettleException;
 import org.pentaho.di.core.row.ValueMeta;
 import org.pentaho.di.core.row.ValueMetaInterface;
+import org.pentaho.di.core.variables.VariableSpace;
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/main/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImpl.java:40]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDataTest.java [hunk 9]
```diff
@@ -21,6 +21,7 @@
 import com.mongodb.util.JSON;
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Matchers;
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDataTest.java:24]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaInjectionTest.java [hunk 10]
```diff
@@ -116,6 +116,24 @@ public boolean get() {
         return meta.isUseSSLSocketFactory();
       }
     } );
+    check( "USE_CONNECTION_STRING", new BooleanGetter() {
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaInjectionTest.java:119]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaTest.java [hunk 11]
```diff
@@ -75,7 +75,7 @@ public static void beforeClass() throws KettleException {
 
           @Override
           public String getTestObject() {
-            return "{" + UUID.randomUUID().toString() + "}";
+            return "{" + UUID.randomUUID() + "}";
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaTest.java:78]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMetaInjectionTest.java [hunk 12]
```diff
@@ -158,6 +158,24 @@ public boolean get() {
         return meta.isUseSSLSocketFactory();
       }
     } );
+    check( "USE_CONNECTION_STRING", new BooleanGetter() {
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMetaInjectionTest.java:161]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputTest.java [hunk 13]
```diff
@@ -1026,7 +1025,7 @@ private void setupRowMeta() {
     Node stepNode = step.item( 0 );
 
     MongoDbOutputMeta newMeta = new MongoDbOutputMeta();
-    newMeta.loadXML( stepNode, (List<DatabaseMeta>) null, (IMetaStore) null );
+    newMeta.loadXML( stepNode, null, (IMetaStore) null );
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputTest.java:1028]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/mongo/wrapper/MongoWrapperUtilTest.java [hunk 14]
```diff
@@ -20,6 +20,8 @@
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/mongo/wrapper/MongoWrapperUtilTest.java:23]: This file needs the copyright date updated.

File: pentaho-mongodb-plugin/src/test/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImplTest.java [hunk 15]
```diff
@@ -42,6 +42,7 @@
 import org.pentaho.di.core.exception.KettlePluginException;
 import org.pentaho.di.core.plugins.PluginRegistry;
 import org.pentaho.di.core.row.value.ValueMetaPluginType;
+import org.pentaho.di.core.variables.VariableSpace;
```
Comment (peterrinehart) [inline pentaho-mongodb-plugin/src/test/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImplTest.java:45]: This file needs the copyright date updated.

**Review by peterrinehart — CHANGES_REQUESTED** [review]
Additional license header changes and style issues need to be fixed; I've tried to call out each file still needing to have the license header updated.
Also, check the number of spaces used to indent some of these blocks; I'm not sure they're consistent with the style guidelines.

**Comment by buildguy** [comment]
### :x: Build failed in 58s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl pentaho-mongodb-plugin
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Pull request merged**